packages/react/src/components/TreeView/TreeNode.tsx

```diff
@@ -288,7 +464,7 @@ TreeNode.propTypes = {
    * Optional prop to allow each node to have an associated icon.
    * Can be a React component class
    */
-  renderIcon: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
```

Comment:
Narrowing proptypes is considered a breaking change. We'll need to leave this as is for now.